### PR TITLE
Yup validation for all text input fields.

### PR DESCRIPTION
**Pull request by JOHNFLEURIMOND**

### CODE REVIEW

File: services-js/commissions-app/src/pages/commissions/apply.tsx [hunk 1]
```diff
@@ -56,6 +56,36 @@ export default class IndexPage extends React.Component<Props, State> {
...
+              lastName: Yup.string()
+                .required('Your Last Name Is Required!')
+                .min(2, 'Your Last Name Needs To Be Valid'),
+              address: Yup.string()
+                .required('Your Address Is Required!')
+                .min(2, 'Your Address Needs To Be Valid'),
+              unit: Yup.number()
+                .required('Your Unit or Apartment Is Required!')
+                .positive()
+                .integer(),
```
Comment (fionawhim) [inline services-js/commissions-app/src/pages/commissions/apply.tsx]: Shouldn't be required, or forced to be a positive integer ("2B" is a valid apartment number, for example).

**Review by fionawhim — CHANGES_REQUESTED** [review]

File: services-js/commissions-app/src/pages/commissions/apply.tsx [hunk 2]
```diff
@@ -59,17 +59,15 @@ export default class IndexPage extends React.Component<Props, State> {
...
                 .min(2, 'Your First Name Needs To Be Valid'),
-
               lastName: Yup.string()
                 .required('Your Last Name Is Required!')
                 .min(2, 'Your Last Name Needs To Be Valid'),
               address: Yup.string()
                 .required('Your Address Is Required!')
                 .min(2, 'Your Address Needs To Be Valid'),
-              unit: Yup.number()
+              unit: Yup.string()
```
Comment (fionawhim) [inline services-js/commissions-app/src/pages/commissions/apply.tsx]: Unit still should not be required.

**Review by fionawhim — CHANGES_REQUESTED** [review]

File: services-js/commissions-app/src/pages/commissions/apply.tsx [hunk 3]
```diff
@@ -52,6 +48,16 @@ export default class IndexPage extends React.Component<Props, State> {
               email: '',
               confirmEmail: '',
             }}
+            validate={values => {
+              // same as above, but feel free to move this into a class method now.
+              let errors = {};
+              if (this.email.value != this.confirmEmail.value) {
```
Comment (fionawhim) [inline services-js/commissions-app/src/pages/commissions/apply.tsx]: hang on. Don't add new stuff to this PR right now. Do the confirm email on a different branch.

**Review by fionawhim — CHANGES_REQUESTED** [review]

File: services-js/commissions-app/src/pages/commissions/apply.tsx (hunk 3 above)
Comment (JOHNFLEURIMOND) [reply services-js/commissions-app/src/pages/commissions/apply.tsx]: Yeah that was by accident. I will correct that now

**Review by fionawhim — APPROVED** [review]
LGTM!

**Pull request merged**